referenceframe/frame_system.go

```diff
@@ -712,3 +712,18 @@ func poseFromPositions(frame Frame, positions map[string][]Input) (spatial.Pose,
 	}
 	return frame.Transform(inputs)
 }
+
+// AddFrameBetween takes an existing framesystem which has general form:
+// originalFS := world - f1 - f2 - ...
+// and places the middle frame between f1 and world resulting in
+// newFS := world - middle - f1 - f2 ...
+func AddFrameBetween(middle Frame, originalFS FrameSystem) (FrameSystem, error) {
```

Comment:
I am not particularly excited by this function, so it not used else where in this PR until I get reviewers' thoughts on how useful this can be. My hesitation stems from this:
Suppose we have the following:
```
originalFS := world - f1 - f2 - ...
```
As it stands this function cannot do:
```
originalFS := world - f1 - middle - f2 - ...
```
i.e. we cannot place the `"middle"` frame anywhere else but between `world` and `f1`.
To accomplish the above, I am pretty sure that we should need to add this as a method to the FrameSystem interface so that then we have access to the children and parents.